hack/api-migration/main.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	networkingv1beta "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	// This package is required to be imported to register all client
+	// plugins.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+const (
+	oldGroupName = "certmanager.k8s.io"
+)
+
+var (
+	kubeconfig string
+	origFile   string
+	newFile    string
+	migrations = map[string]string{
+		"certmanager.k8s.io/acme-http01-edit-in-place": "acme.cert-manager.io/http01-edit-in-place",
+		"certmanager.k8s.io/acme-http01-ingress-class": "acme.cert-manager.io/http01-ingress-class",
+
+		"certmanager.k8s.io/issuer":           "cert-manager.io/issuer",
+		"certmanager.k8s.io/cluster-issuer":   "cert-manager.io/cluster-issuer",
+		"certmanager.k8s.io/alt-names":        "cert-manager.io/alt-names",
+		"certmanager.k8s.io/ip-sans":          "cert-manager.io/ip-sans",
+		"certmanager.k8s.io/common-name":      "cert-manager.io/common-name",
+		"certmanager.k8s.io/issuer-name":      "cert-manager.io/issuer-name",
+		"certmanager.k8s.io/issuer-kind":      "cert-manager.io/issuer-kind",
+		"certmanager.k8s.io/certificate-name": "cert-manager.io/certificate-name",
+	}
+
+	depreciations = []string{
+		"certmanager.k8s.io/acme-challenge-type",
+		"certmanager.k8s.io/acme-dns01-provider",
+	}
+)
+
+var cmd = &cobra.Command{
+	Use:  "cert-manager-api-migration",
+	Long: "A tool to download Ingress resources and convert cert-manager annotations from certmanager.k8s.io to cert-manager.io. This tool will not modify resources in the API server.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(kubeconfig) == 0 {
+			kubeconfig = os.Getenv("KUBECONFIG")
+		}
+
+		if len(kubeconfig) == 0 {
+			return errors.New("flag --kubeconfig or environment variable $KUBECONFIG must be set")
+		}
+
+		client, err := clientSet(kubeconfig)
+		if err != nil {
+			return err
+		}
+
+		ingList, err := client.NetworkingV1beta1().Ingresses("").List(metav1.ListOptions{})
```

Comment:
Best to use ExtensionsV1beta1 here - networking v1beta1 only introduced Ingress in k8s 1.15 iirc